src/commands/authentication/UserLoginPersistent.ts

```diff
@@ -0,0 +1,231 @@
+                return;
+            }
+            if (verifiedRole.members.has(buttonInteraction.user.id)) {
+                buttonInteraction.reply({ content: "Your account is already linked to an email in our system. To link your discord account to another email address, call `/logout` first.", ephemeral: true });
+                return;
+            }
+
+
+            if (buttonInteraction.customId == `logInWithEmail-${interaction.id}`) {
+                const modalInputEmail: string = await this.displayLogInWithEmailModal(buttonInteraction);
+                if (await this.isExistingEmail(modalInputEmail)) {
+                    this.distributeCode(buttonInteraction.user.id, modalInputEmail);
+                    buttonInteraction.followUp({ content: `Please check ${modalInputEmail} for the verification code we just sent you.`, ephemeral: true })
+                } else {
+                    buttonInteraction.followUp({ content: "**We couldn't find the email address in our system. Please try again.**", ephemeral: true });
+                }
+            } 
+            
+            else if (buttonInteraction.customId == `verification-${interaction.id}`) {
+                //FIXME: rework how client email db table works, 
+                const modalInputEmail: string | undefined = await this.getUserEmail(buttonInteraction.user.id);
+                if (!modalInputEmail) {
+                    buttonInteraction.reply({ content: "**Please input your email using the button to the left first before verifying.**", ephemeral: true });
+                    return;
+                }
+
+                const modalInputVerificationCode: string = await this.displayVerificationModal(buttonInteraction);
+                if (await this.verifyCode(buttonInteraction.user.id, modalInputVerificationCode)) {
+                    await ClientEmail.destroy({where: {discord_client_id: buttonInteraction.user.id}});
+                    await ClientCode.destroy({where: {discord_client_id: buttonInteraction.user.id}});
+                    this.assignVerifiedRole(interaction);
+                    buttonInteraction.followUp({ content: `**Verification successful**. Your discord account is now linked to ${modalInputEmail}!`, ephemeral: true });
+                } else {
+                    buttonInteraction.followUp({ content: `**Verification failed**. Please try entering the email or the verification code again.`, ephemeral: true });
+                }
+            }
+        })
+    }
+
+
+
+    private async displayLogInWithEmailModal(interaction: ButtonInteraction): Promise<string> {
+        const emailModal: ModalBuilder = new ModalBuilder()
+        .setCustomId(`login-${interaction.id}`)
+        .setTitle("Log in");
+
+        const emailInput: TextInputBuilder = new TextInputBuilder()
+        .setCustomId("emailInput")
+        .setLabel("Please enter the registered email address.")
+        .setValue("E.G. f100lastname@uwaterloo.ca")
```

Comment:
If this requires the person to CTRL-A + Backspace to get rid of this input, then lets just remove it. If its a placeholder that goes away when you click on the textbox, then keep it.